### PR TITLE
Ch 1860: Page layout changes for new campaign wizard flow

### PR DESCRIPTION
**Pull request by eshtadc**

### CODE REVIEW

File: personalize.admin.campaign.inc [hunk 1]
```diff
@@ -0,0 +1,1447 @@
...
+    $form += $function($form, $form_state, $agent_data, $agent_instance);
+  }
+
+  // Add the form action buttons.
+  $form += personalize_campaign_wizard_action($form, $form_state, $agent_data, $agent_instance);
+  return $form;
+}
+
+/**
+ * Generate the multi-step form navigation display.
```
Comment (eshtadc) [inline personalize.admin.campaign.inc:86]: Should indicate that this is the "process bar" - not just navigation.

File: personalize.admin.inc [hunk 2]
```diff
@@ -915,120 +750,11 @@ function personalize_agent_date_form_submit($agent_name, $values) {
...
-          $agent_data->machine_name => array(
-            'status' => $current_status,
-            'nextStatus' => array(
-              'status' => $next_status,
-              'text' => $status_text,
-            ),
-          ),
-        ),
-      ),
-    );
```
Comment (katbailey) [inline personalize.admin.inc:1160]: Yay :) (Although this also means your branch isn't based off the latest RC - I think you have some fun rebasing in your future :/)

File: personalize.api.php [hunk 3]
```diff
@@ -56,15 +56,14 @@ function hook_personalize_option_set_type() {
...
- * modules can alter the information that is saved.
+ * Allows the alteration of an agent before it is saved.
  *
  * @param stdClass $agent
  *   A stdClass object representing the agent being saved.
+ * @param array $prior_values
+ *   The unalterable prior data for the agent before this save.
  */
-function hook_personalize_agent_presave($agent) {
+function hook_personalize_agent_alter(&$agent, $prior_values) {
```
Comment (katbailey) [inline personalize.api.php]: Hmm, presave seems more like the right hook here, why the change to _alter?

Comment (eshtadc) [inline personalize.api.php]: Well - first presave actually wasn't being called in the personalize module anyway.  But when I looked at adding it in I remembered that module_invoke passes by value and not by reference.

File: personalize.admin.inc (hunk 2 above)
Comment (eshtadc) [inline personalize.admin.inc:1160]: Yeah - our timing on these was unfortunate.

File: personalize.api.php (hunk 3 above)
Comment (katbailey) [inline personalize.api.php]: Objects are always passed by reference. That is weird that we're no longer invoking hook_personalize_agent_presave() especially seeing as there's an implementation of it in acquia_lift :P Anyway, _alter just feels wrong for this as it's usually something you'd use for altering something that's about to be used, not something that's about to be saved to the db.

Comment (eshtadc) [inline personalize.api.php]: The information that was in the presave hook didn't matter anyway cause it was being set as defaults later on.  But this will matter.  I'll put it back to presave and actually invoke it then.  Of course an object is by reference - that fixes it all :-)

File: personalize.admin.campaign.inc [hunk 4]
```diff
@@ -0,0 +1,1475 @@
...
+ * @param stdClass $agent_data
+ *   The current campaign data.
+ * @param PersonalizeAgentInterface $agent_instance
+ *   The agent that defines the functionality for the campaign (if set).
+ * @return array
+ *   A renderable array of instructions.
+ */
+function personalize_campaign_wizard_instructions_scheduling($form, &$form_state, $agent_data, $agent_instance = NULL) {
+  return array(
+    '#markup' => 'Pellentesque habitant morbi tristique senectus Pellentesque habitant morbi tristique senectus Pellentesque habitant morbi tristique senectus Pellentesque habitant morbi tristique senectus.',
```
Comment (katbailey) [inline personalize.admin.campaign.inc:1072]: :)
